### ISSUE TITLE
Test

### ISSUE THREAD

**Issue by DAMSCROW** [DAMSCROW/Misham#2]
Storybook is a UI component development tool used to design, develop, and test individual components in isolation from your main app. It helps developers build and showcase components without running the entire app, which allows for faster iteration and better organization of components. In React Native apps, using Storybook helps you develop UI elements independently, ensuring they are modular, reusable, and well-tested.

From the image of your project directory and code changes, it seems that Storybook has been integrated into your React Native project with the Expo framework. Here's a detailed explanation of what the different files are doing as part of this integration:

Key Storybook-related Files and Changes:

1. .storybook Directory:

This directory usually holds Storybook's configuration files.

Files like main.ts, preview.tsx, and storybook.requires.ts are crucial in configuring how Storybook works with your app.

main.ts defines the structure and where Storybook looks for stories (like specific components).

preview.tsx can contain global decorators or parameters for stories (e.g., theming).

storybook.requires.ts registers all the components (stories) to be showcased.

2. src/Button.stories.tsx:

This is where the actual "story" for a component (like Button.tsx) is defined.

Stories provide various test cases or usage scenarios for a component. For example, the story may define how the button should render in different states (disabled, clicked, hovered, etc.).

The .stories.tsx files are meant to separate component logic from how the component is used and displayed.

3. Changes in package.json:

Storybook-related dependencies, like @storybook/react-native, are likely being added.

Scripts to run Storybook locally in your app would also be updated. For example, a new npm script like start-storybook could have been added.

4. assets/ directory:

The assets folder holds images and icons that might be necessary for displaying components in Storybook (like favicon.png, icon.png, etc.).

5. Other configuration files (babel.config.js, tsconfig.json):

Storybook may need modifications in your Babel and TypeScript configurations to work seamlessly with your app's components.

For instance, adding polyfills, aliases, or plugins to support Storybook’s functionality with React Native and Expo.

6. App.tsx:

You might have conditional logic in App.tsx to launch Storybook instead of the normal app, depending on the environment. For example, if a specific environment variable (like STORYBOOK_ENABLED) is set, it could boot up the Storybook instead of the actual application.

7. Changes to the Button.tsx component:

The Button.tsx component is modified to be compatible with Storybook or to ensure it works well in an isolated environment.

These changes are likely minor adjustments, like ensuring that necessary props are passed and supported when used independently in a story format.

Purpose of Introducing Storybook:

Component Isolation: Storybook helps test UI components in isolation without running the entire app, which can be time-consuming.

UI Documentation: By writing stories, you're essentially documenting the use cases for components. This is useful for team collaboration and understanding how a component behaves across different states.

Visual Testing: It allows for visual testing of components by easily comparing component states or how they appear when different props are applied.

Development Speed: Faster development cycles as you can work on a single component without worrying about the broader app environment.

Design System: Storybook acts as a centralized place for developing a design system, making sure that the UI components are reusable and consistent across the app.

In summary, the changes introduced by Storybook are aimed at improving the development, testing, and visualization of UI components within your Expo-based React Native app. With Storybook, developers and designers can iterate faster on components, ensuring they work as expected before integrating them into the larger app.

**Comment by DAMSCROW** [comment]
Here’s a refined version of the problem description you can use for your start-up call:


---

During the process of implementing the ConstructInternalRouteWhiteListRule, we encountered an issue with how it interacted with an existing rule. The objective of this new rule was to restrict access to internal IPs when a specific route containing /internal/* was detected in the request path.

Initially, we had an existing rule in place that only allowed access to APIs if the request came from an internal route, effectively blocking all other external routes. However, as part of our testing, this rule was expected to be removed as the application moves to production.

When I attempted to test the newly created rule by removing the existing rule, we encountered a problem. Since the Web Application Firewall (WAF) has a default action of blocking all incoming requests unless explicitly allowed, without the existing rule, all incoming requests were blocked. The new rule alone was insufficient to process requests because it was set up to allow access based on specific IPs and paths, but no complementary rule existed to allow other non-internal routes.

This situation led us to realize that we needed to create two distinct rules:

1. One rule that allows non-internal routes, bypassing the need for IP checking.


2. Another rule that permits access to internal routes only if the request originates from an approved IP set.



These two rules would ensure a balance between internal and external access, with the proper fallback to block unwanted requests based on specific conditions.


---

This explanation provides clarity on the situation and decision-making process regarding the WAF rule setup and the need for the two new rules. Let me know if any adjustments are needed!

**Comment by DAMSCROW** [comment]
From the image, the changes in the button.tsx file appear to be the creation of a new button component in React Native using TypeScript. Below is a detailed explanation based on the visible code:

Code Breakdown:

1. Imports:

import React from 'react';
import { Pressable, Text, StyleSheet } from 'react-native';

The component imports React and essential UI components from react-native such as Pressable (for handling button presses) and Text (for displaying text inside the button).

StyleSheet is used to create a consistent styling system within the component.



2. Styles Definition:

const styles = StyleSheet.create({
  container: {
    paddingHorizontal: 32,
    paddingVertical: 8,
    backgroundColor: 'purple',
    alignSelf: 'flex-start',
    borderRadius: 4,
  },
  text: {
    color: 'white',
    fontSize: 16,
    fontWeight: 'bold',
  },
});

The StyleSheet.create function defines a container style for the button and a text style for the text inside the button.

container:

paddingHorizontal: 32: Horizontal padding to give some space on the sides of the button.

paddingVertical: 8: Vertical padding to create space above and below the text inside the button.

backgroundColor: 'purple': Sets the button’s background color to purple.

alignSelf: 'flex-start': Aligns the button to the start of the container it is placed in.

borderRadius: 4: Rounds the button's corners.


text:

color: 'white': The text color inside the button is white.

fontSize: 16: The size of the text is set to 16.

fontWeight: 'bold': Makes the text bold.




3. Type Definition for Button Props:

export type ButtonProps = {
  onPress: () => void;
  text: string;
};

The ButtonProps type defines the expected props for this button component:

onPress: A function that is triggered when the button is pressed.

text: A string to be displayed inside the button.




4. Button Component Definition:

const Button = ({ onPress, text }: ButtonProps) => (
  <Pressable
    style={styles.container}
    onPress={onPress}
  >
    <Text style={styles.text}>{text}</Text>
  </Pressable>
);

The button component uses the Pressable component, which is styled with styles.container.

The onPress prop is passed to the Pressable to handle press actions.

Inside the Pressable, a Text component displays the text prop, which is styled using styles.text.




Summary of Changes:

The code introduces a simple button component using Pressable and Text.

The button is styled with custom styles for padding, background color, and text properties.

The component is designed to accept onPress (a callback function for handling button press actions) and text (the label to display inside the button) as props.

The button can be reused anywhere in the application by passing different text and actions through props.


Let me know if you need further clarification!

**Comment by DAMSCROW** [comment]
From the image, the changes in the button.tsx file appear to be the creation of a new button component in React Native using TypeScript. Below is a detailed explanation based on the visible code:

Code Breakdown:

1. Imports:

import React from 'react';
import { Pressable, Text, StyleSheet } from 'react-native';

The component imports React and essential UI components from react-native such as Pressable (for handling button presses) and Text (for displaying text inside the button).

StyleSheet is used to create a consistent styling system within the component.



2. Styles Definition:

const styles = StyleSheet.create({
  container: {
    paddingHorizontal: 32,
    paddingVertical: 8,
    backgroundColor: 'purple',
    alignSelf: 'flex-start',
    borderRadius: 4,
  },
  text: {
    color: 'white',
    fontSize: 16,
    fontWeight: 'bold',
  },
});

The StyleSheet.create function defines a container style for the button and a text style for the text inside the button.

container:

paddingHorizontal: 32: Horizontal padding to give some space on the sides of the button.

paddingVertical: 8: Vertical padding to create space above and below the text inside the button.

backgroundColor: 'purple': Sets the button’s background color to purple.

alignSelf: 'flex-start': Aligns the button to the start of the container it is placed in.

borderRadius: 4: Rounds the button's corners.


text:

color: 'white': The text color inside the button is white.

fontSize: 16: The size of the text is set to 16.

fontWeight: 'bold': Makes the text bold.




3. Type Definition for Button Props:

export type ButtonProps = {
  onPress: () => void;
  text: string;
};

The ButtonProps type defines the expected props for this button component:

onPress: A function that is triggered when the button is pressed.

text: A string to be displayed inside the button.




4. Button Component Definition:

const Button = ({ onPress, text }: ButtonProps) => (
  <Pressable
    style={styles.container}
    onPress={onPress}
  >
    <Text style={styles.text}>{text}</Text>
  </Pressable>
);

The button component uses the Pressable component, which is styled with styles.container.

The onPress prop is passed to the Pressable to handle press actions.

Inside the Pressable, a Text component displays the text prop, which is styled using styles.text.




Summary of Changes:

The code introduces a simple button component using Pressable and Text.

The button is styled with custom styles for padding, background color, and text properties.

The component is designed to accept onPress (a callback function for handling button press actions) and text (the label to display inside the button) as props.

The button can be reused anywhere in the application by passing different text and actions through props.


Let me know if you need further clarification!

M

**Comment by DAMSCROW** [comment]
To provide a detailed update for your story, you can expand on your existing notes regarding the Storybook implementation with specific details from the current Button.stories.tsx and Button.tsx files. Here’s how you can structure your update based on your analysis:


---

Story Update on Storybook Implementation in Expo-based React Native Application

"As part of this story, I have reviewed PRs from the Veracle team regarding the integration of Storybook and its impact on our default dashboard behavior. My analysis focused on understanding the frameworks used and how Storybook aids in UI component development. Storybook serves as a valuable tool for building, testing, and visualizing components in isolation, which enhances modularity and allows for faster iteration. This integration is particularly beneficial for our Expo-based React Native app, enabling more efficient development and testing of reusable UI elements."

Current Storybook Implementation - Button Component Overview:

We have implemented Storybook within our application, focusing on defining stories that showcase the behavior of individual UI components in isolation. As part of this, the Button.stories.tsx file has been created to define basic scenarios for our button component.

Key Implementation Details:

Component Metadata: In Button.stories.tsx, the metadata for the Button component is defined, linking it to the Storybook environment under the title 'Button'. This ensures that the component is properly registered and can be viewed and tested in isolation from the rest of the app.

Basic Story Definition: A simple story is provided to demonstrate the basic behavior of the button. It showcases:

Button text: "Hello world"

onPress action: Logs a message ("Button clicked") when pressed


This story allows developers to visualize and interact with the basic functionality of the button without needing to run the entire app.

const Basic: Story = {
  args: {
    text: 'Hello world',
    onPress: () => console.log('Button clicked'),
  },
};

Component Styling and Functionality (Button.tsx): The button component itself (Button.tsx) is styled using the StyleSheet API from React Native. It accepts two key props:

onPress: A callback function triggered when the button is pressed.

text: The label to be displayed inside the button.


This setup ensures flexibility, allowing the button component to be reused across various parts of the application with different labels and actions.

const Button = ({ onPress, text }: ButtonProps) => (
  <Pressable
    style={styles.container}
    onPress={onPress}
  >
    <Text style={styles.text}>{text}</Text>
  </Pressable>
);


Importance of This Implementation:

The introduction of Storybook and the creation of basic stories, such as the one for the button component, provide several key benefits:

Component Isolation: Developers can now interact with the Button component in isolation, focusing on its behavior without worrying about how it interacts with other parts of the app.

Faster Iteration: As changes to the button component are made, Storybook allows for rapid testing and visualization of those changes without needing to restart the full application.

Improved Collaboration: Storybook also acts as a form of documentation, allowing team members (both developers and designers) to see how components behave under different conditions and states.


Next Steps:

As we continue to enhance the Storybook setup, we will add more stories to reflect different states of the button, such as disabled, loading, and variant styles (e.g., primary, secondary). This will further strengthen the reusability and testability of our UI components.


---

This detailed update can be submitted as part of your story, outlining both the current state of the Storybook integration and the specific implementation of the Button component in your application.

**Comment by DAMSCROW** [comment]
Here’s a refined and professional summary for your status update based on the image and your explanation:


---

Status Update for EMBER-324:

As part of the task to create a new internal route and move the health check endpoint to it, I have completed the following:

1. API Code:

Created a new /internal route in a separate internal.ts file.

Moved the existing health check API into the internal HONO app.



2. Infrastructure (CDK) Code:

Initially, the plan was to introduce an explicit WAF rule to restrict access to the /internal route for the Salal internal network. However, we already had a WAF rule that restricted all APIs to internal access.

After discussions with Chris and the team, we decided to have separate rules for non-production and production environments:

Non-Production Environment: We will keep the existing WAF rule that blocks any incoming request by default unless the request IP is part of the internal Salal network. No new rule will be introduced here.

Production Environment:

We introduced a specific WAF rule that restricts access to /internal routes, allowing requests only from internal IPs.

Additionally, a temporary rule has been added to block any external IPs for /internal URLs during the production migration. This temporary rule will be removed once the migration is complete.






The environment has been modified to reflect these changes, and the necessary WAF restrictions are now in place for both environments.


---

Let me know if any additional details are needed!

**Comment by DAMSCROW** [comment]
Here’s a refined and professional summary for your status update based on the image and your explanation:


---

Status Update for EMBER-324:

As part of the task to create a new internal route and move the health check endpoint to it, I have completed the following:

1. API Code:

Created a new /internal route in a separate internal.ts file.

Moved the existing health check API into the internal HONO app.



2. Infrastructure (CDK) Code:

Initially, the plan was to introduce an explicit WAF rule to restrict access to the /internal route for the Salal internal network. However, we already had a WAF rule that restricted all APIs to internal access.

After discussions with Chris and the team, we decided to have separate rules for non-production and production environments:

Non-Production Environment: We will keep the existing WAF rule that blocks any incoming request by default unless the request IP is part of the internal Salal network. No new rule will be introduced here.

Production Environment:

We introduced a specific WAF rule that restricts access to /internal routes, allowing requests only from internal IPs.

Additionally, a temporary rule has been added to block any external IPs for /internal URLs during the production migration. This temporary rule will be removed once the migration is complete.






The environment has been modified to reflect these changes, and the necessary WAF restrictions are now in place for both environments.


---

Let me know if any additional details are needed!

/**
 * Internal Health Check API
 * 
 * This API has been moved to an internal route and is only accessible by internal ZALAL users.
 * Route: GET /healthcheck
 */

**Comment by DAMSCROW** [comment]
Here’s a refined and professional summary for your status update based on the image and your explanation:


---

Status Update for EMBER-324:

As part of the task to create a new internal route and move the health check endpoint to it, I have completed the following:

1. API Code:

Created a new /internal route in a separate internal.ts file.

Moved the existing health check API into the internal HONO app.



2. Infrastructure (CDK) Code:

Initially, the plan was to introduce an explicit WAF rule to restrict access to the /internal route for the Salal internal network. However, we already had a WAF rule that restricted all APIs to internal access.

After discussions with Chris and the team, we decided to have separate rules for non-production and production environments:

Non-Production Environment: We will keep the existing WAF rule that blocks any incoming request by default unless the request IP is part of the internal Salal network. No new rule will be introduced here.

Production Environment:

We introduced a specific WAF rule that restricts access to /internal routes, allowing requests only from internal IPs.

Additionally, a temporary rule has been added to block any external IPs for /internal URLs during the production migration. This temporary rule will be removed once the migration is complete.






The environment has been modified to reflect these changes, and the necessary WAF restrictions are now in place for both environments.


---

Let me know if any additional details are needed!

/**
 * Internal Health Check API
 * 
 * This API has been moved to an internal route and is only accessible by internal ZALAL users.
 * Route: GET /healthcheck
 */

/**
 * Constructs a WAF rule to restrict access to internal routes based on a whitelist of authorized IPs.
 * 
 * @param {string} internalIPSetArn - The ARN of the internal IP set used for whitelisting.
 * @param {number} priority - The priority of the WAF rule.
 * @returns {CfnWebACL.RuleProperty} The WAF rule that restricts access to internal routes for unauthorized IPs.
 * 
 * This rule blocks all requests to routes starting with `/internal`, unless the request originates
 * from an IP address within the specified internal IP set.
 */

**Comment by DAMSCROW** [comment]
/**
 * Constructs a WAF (Web Application Firewall) global rule for blocking non-sail IPs in the production environment.
 * 
 * @param ipSetArn - The ARN (Amazon Resource Name) of the IP Set to reference.
 * @param priority - The priority of the rule within the WAF. Lower values indicate higher priority.
 * @returns A `CfnWebACL.RuleProperty` object defining the rule properties for WAF.
 *
 * The rule:
 * - Blocks requests originating from IPs not included in the specified IP Set (non-sail IPs).
 * - Configures visibility settings to enable sampling and CloudWatch metrics for monitoring purposes.
 * - Uses the `notStatement` to allow only requests from the given IP Set.
 */
export function constructProductionGlobalBlockRule(
  ipSetArn: string,
  priority: number
): CfnWebACL.RuleProperty {
  return {
    name: 'BlockNonSailIPs',
    priority: priority,
    action: { block: {} },
    visibilityConfig: {
      sampledRequestsEnabled: true,
      cloudWatchMetricsEnabled: true,
      metricName: 'Metric-BlockNonSailIPs',
    },
    statement: {
      notStatement: {
        statement: {
          ipSetReferenceStatement: {
            arn: ipSetArn,
          },
        },
      },
    },
  };
}

**Comment by DAMSCROW** [comment]
// Define the IP set for the WAF configuration
const ipSet = createWafIpSet(
  this,
  CLOUDFRONT,
  `${PHOENIX_IP_RULESET}ForFrontEnd`
);

// Internal route whitelist rule to allow traffic only from trusted internal sources
const internalIPWhitelistRule = constructInternalRouteWhitelistRule(
  ipSet.attrArn,
  2
);

// Temporary "block non-SLAL IP" rule to restrict API access to internal SLAL IPs only.
// This rule ensures that no external IP can access the APIs before the application goes live.
// Once the application is live, this rule will be removed, and the API will be open to all users.
const blockNonSLAIPRule = constructProductionGlobalBlockRule(ipSet.attrArn, 3);

// Allow IP rule to permit traffic for non-production environments
const allowIPRule = constructIpWhitelistRule(ipSet.attrArn, 3);

// Production ruleset: Includes strict internal whitelist and the temporary block non-SLAL IP rule
const frontEndWafRulesForProd: CfnWebACL.RuleProperty[] = [
  internalIPWhitelistRule,
  blockNonSLAIPRule, // Temporary rule to be removed at production go-live
  ...emberFrontEndWAFRules,
];

// Non-production ruleset: Includes more relaxed rules, such as allow list for testing purposes
const frontEndWafRulesForNonProd: CfnWebACL.RuleProperty[] = [
  allowIPRule, // Allow rule to facilitate testing in non-production
  ...emberFrontEndWAFRules,
];

// Identify the environment based on the stack configuration (production or non-production)
const envName = this.stackConfig.account.name;

// Create the WAF with the appropriate rules based on the environment
this.waf = createWAF({
  stack: this,
  wafName: 'FrontEndWaf',
  scope: CLOUDFRONT,
  webACLProps: {
    defaultAction: envName === 'production' ? { allow: {} } : { block: {} },
    rules: envName === 'production'
      ? frontEndWafRulesForProd
      : frontEndWafRulesForNonProd,
  },
});

**Comment by DAMSCROW** [comment]
Comprehensive Overview: Transitioning from React Web Application to React Native

Our goal is to transform our existing React web application into a codebase that supports both web and mobile platforms using React Native. This transition is facilitated through the Expo framework, allowing us to create an efficient, shared codebase. Below is a detailed breakdown of the current application structure, the expectations for the transition, and the specific steps and guidance provided by Sam.


---

1. Current Project Status

We are starting with a React web application. The aim is to extend this application to support mobile platforms using React Native.

The transition will leverage Expo, which simplifies the development process for both web and mobile applications.

Our toolset includes Storybook for component testing and GlueStack UI for building modular UI components, though we are currently focusing on React Native components to avoid delays.



---

2. Component Structure and Organization

General Principles

The new codebase will focus on creating a shared code structure for maximum reusability, while still allowing for platform-specific implementations where necessary.

The structure will aim for cognitive consistency, meaning components should be organized in a way that is intuitive and easy to navigate.


Proposed Component Structure

Folder Structure:

/components
    /Card
        Card.tsx (generic card component)
        Card.web.tsx (if web-specific styles needed)
        Card.native.tsx (if mobile-specific styles needed)
    /Background
        BlueBackground.tsx (specific to this layout)
    /Header
        Header.tsx (for app bar/header component)
    /Screens
        Home.tsx (main screen combining components)
        Authentication.tsx (authentication specific screen)


Handling Layouts

Shared vs. Platform-Specific: Most components will share logic, but some might need platform-specific implementations using .web.tsx and .native.tsx file extensions.

Blue Background and Card Component:

Create a generic Card component that can be reused across various screens.

If the blue background is unique to the page but potentially reusable, create a BlueBackground.tsx component. If it’s only for the current layout, apply styles directly within the Home.tsx screen.




---

3. Detailed Component Creation Guidelines

Blue Background

If Reusable:

Create a BlueBackground.tsx file that defines the styling for the blue background (e.g., a deep navy color).

This file should export a styled component that can be imported into other screens.


If Not Reusable:

Directly style the blue background within the Home.tsx file.



Card Component

Implement a generic card component in Card.tsx, which can accept props for content and styling.

If there are specific styling needs for web or mobile, create respective files Card.web.tsx and Card.native.tsx to handle these variations.


Header Component

Header Structure:

The header should be a separate component that includes elements like the logout button and any branding (logo).

This header will be part of the Compound UI library, ensuring it’s accessible across the application.


Ticket Structure for Header Creation:

Create separate tickets for various aspects of the header:

Creating the header layout and styling.

Implementing the logout button functionality.

Ensuring the header is responsive across different screen sizes.




Using Conditional Logic for Platform-Specific Designs

Utilize React Native's Platform module to conditionally render elements or apply styles based on whether the app is running on the web or mobile. For example:

import { Platform } from 'react-native';

const styles = {
    container: {
        backgroundColor: Platform.OS === 'web' ? 'blue' : 'navy',
        padding: 10,
    },
};



---

4. Integration of Storybook for Component Testing

Launching Storybook:

Use the command pnpm web to start Storybook, which allows you to see components rendered in isolation.

You can navigate through different stories for each component to ensure they function correctly and meet design specifications.


Debugging:

If you encounter issues with components (e.g., missing dependencies), you can address these in isolation within Storybook, making it easier to identify and fix problems.


Future Mobile Testing:

Once development environments (e.g., MacBooks) are set up, you will be able to test components on simulators or emulators for both iOS and Android.




---

5. Development Workflow and Component Organization

Managing Complexity in Tickets

Separate Tickets for Components:

Each component should have its own ticket to ensure clarity in tasks and responsibilities.

Avoid combining multiple component creations into a single ticket, as this can lead to confusion and overwhelm.


Iterative Development:

Initially create stubs or basic versions of components. This allows for a foundational structure while leaving room for refinements and enhancements later.

Focus on getting the structure right first. For example, the header might initially just have a placeholder for the logout button, which can be implemented later.



Scoping Tickets Appropriately

When defining tickets, clearly state the scope of work. For example:

Header Component Ticket: Should cover the layout, responsiveness, and inclusion of specific buttons.

Card Component Ticket: Should focus on creating the card with props and styling.




---

6. Example Use Case: Home Screen Structure

Home.tsx Implementation

Combine Components:

Import the BlueBackground and Card components into the Home.tsx file.

Structure the layout so the background wraps the card.



import React from 'react';
import { View } from 'react-native';
import BlueBackground from '../components/Background/BlueBackground';
import Card from '../components/Card/Card';

const Home = () => {
    return (
        <BlueBackground>
            <View>
                <Card title="Welcome!" content="This is your home screen." />
            </View>
        </BlueBackground>
    );
};

export default Home;


---

7. Long-Term Vision and Collaborative Support

Shared Codebase:

The end goal is to have a consistent and maintainable codebase that effectively supports both web and mobile functionalities.

Keep the communication lines open. Sam and others are available to assist with any questions or guidance needed as you navigate through this process.


Incremental Improvements:

Acknowledge that initial designs may not be perfect. The focus should be on functionality and structure first. Styles and components can be iteratively refined through pull requests and feedback.




---

8. Closing Remarks and Future Steps

Next Steps:

Start by implementing the suggested component structure.

Use Storybook to test components as they are created.

Break down tasks into manageable tickets to keep the development process smooth.


Communication:

If any issues arise during implementation, or if there are questions about component use or styling, don’t hesitate to reach out for assistance. This phase is about learning and refining processes as a team.



By following this detailed plan, we will successfully transition our React web application to a React Native codebase that supports both web and mobile functionalities, ensuring a smooth development experience and cohesive user interface across platforms.

Attached is the existing front end application structure make the explanation more details with the specimen

**Comment by DAMSCROW** [comment]
Here are three bullet points based on your requirements, focusing on cloud infrastructure utilization, backend security, and MVP delivery:


---

Project Ember – New Group Digital Platform

Emphasis on optimizing cloud infrastructure with a focus on maximizing AWS feature usage, ensuring enhanced security for backend APIs, and incorporating advanced cloud services for scalability and resilience.

Collaboration between Delivery, Engineering, and Veracruz mobile teams, with backend development aligned with React Native UI for mobile app integration, targeting successful MVP delivery by December.

Ongoing efforts to improve the architecture by leveraging client-specific cloud features, ensuring seamless delivery and system evolution towards a robust digital platform.



---

These points align with the structure from the screenshot you shared. Let me know if you need any further modifications!

**Comment by DAMSCROW** [comment]
Here is an updated version with more emphasis on the MVP delivery:


---

Project Ember – New Group Digital Platform

The primary focus is on delivering the MVP by December, with full integration of cloud infrastructure and AWS features, ensuring a secure and scalable backend to support the mobile application.

Close collaboration between the Delivery, Engineering, and Veracruz mobile teams, aligning backend API development and the new React Native UI for a successful and timely MVP release.

Architecture improvements are ongoing, utilizing client-specific cloud features to ensure optimal performance and seamless deployment of the MVP, with future scalability in mind.



---

This revision highlights the MVP delivery as the main priority. Let me know if further adjustments are needed!

**Comment by DAMSCROW** [comment]
Here’s an updated version of the message that includes your confirmation about focusing on screen development rather than component details:


---

Subject: Clarification on Emper Landing Page Development and Expo Framework Integration

Hi [Vendor Team Name],

I hope this message finds you well. We’re currently working on a story to develop a new landing page for the Emper app that will display a welcome message with the customer’s name upon login to enhance the user experience.

At the moment, our application consists of two screens—the Home and Explore pages—organized within the app/tabs folder (home.tsx and explore.tsx). We’re planning to develop the new Emper landing page, and based on the current setup, it seems like it should also reside in the app/tabs folder. Could you kindly confirm if this is the best place for the new screen, or should we follow a different structure as per the Expo framework guidelines?

We also wanted to confirm the expectation from our side: we will be focusing primarily on developing the basic screen in the format provided. The detailed UI features, styling, and more advanced components will be provided by your team, as we understand you’ll be handling the component development. Our role is more focused on the screen development and integrating it into the app's flow, ensuring the functionality works as expected. Could you confirm that this aligns with your expectations?

Additionally, since we’re just taking our first steps in React Native development, we were wondering if you could kindly provide any documentation or references for guidelines on best practices in using the Expo framework and routing. We want to ensure that our work aligns with yours and meets the framework’s standards.

We apologize if any of our questions seem basic, but we greatly appreciate your guidance as we navigate this process. We’re keen to learn and collaborate effectively to make this a smooth experience for everyone involved.

Thank you so much for your support, and we look forward to your advice!

Best regards,
[Your Name]
[Your Position/Team]


---

This version clarifies the roles, expectations, and politely requests confirmation about what each team will handle. It remains humble and invites documentation or guidance to ensure alignment with their standards.

**Comment by DAMSCROW** [comment]
import React from 'react';
import { render, waitFor } from '@testing-library/react-native';
import RootLayout from './_layout'; // Adjust the import path as per your project structure
import { QueryClient } from '@tanstack/react-query';

jest.mock('expo-font', () => ({
  useFonts: jest.fn(() => [true]), // Mock font loading as successful
}));

jest.mock('expo-splash-screen', () => ({
  preventAutoHideAsync: jest.fn(),
  hideAsync: jest.fn(),
}));

jest.mock('@components/header/Header', () => () => <Text>Mock Header</Text>); // Mocking Header Component

describe('RootLayout', () => {
  it('renders the layout correctly', async () => {
    const queryClient = new QueryClient();
    const { getByText } = render(<RootLayout />);

    // Wait for fonts to load
    await waitFor(() => {
      expect(getByText('Mock Header')).toBeTruthy();
    });

    // Check if the footer placeholder text is rendered
    expect(getByText('Placeholder for Footer App Bar')).toBeTruthy();
  });

  it('hides the splash screen once fonts are loaded', async () => {
    const hideAsyncMock = require('expo-splash-screen').hideAsync;

    render(<RootLayout />);

    // Ensure splash screen hides after font loading
    await waitFor(() => {
      expect(hideAsyncMock).toHaveBeenCalled();
    });
  });
});

**Comment by DAMSCROW** [comment]
import React from 'react';
import { render, waitFor } from '@testing-library/react-native';
import RootLayout from './_layout'; // Adjust the import path as per your project structure
import { QueryClient } from '@tanstack/react-query';

jest.mock('expo-font', () => ({
  useFonts: jest.fn(() => [true]), // Mock font loading as successful
}));

jest.mock('expo-splash-screen', () => ({
  preventAutoHideAsync: jest.fn(),
  hideAsync: jest.fn(),
}));

jest.mock('@components/header/Header', () => () => <Text>Mock Header</Text>); // Mocking Header Component

describe('RootLayout', () => {
  it('renders the layout correctly', async () => {
    const queryClient = new QueryClient();
    const { getByText } = render(<RootLayout />);

    // Wait for fonts to load
    await waitFor(() => {
      expect(getByText('Mock Header')).toBeTruthy();
    });

    // Check if the footer placeholder text is rendered
    expect(getByText('Placeholder for Footer App Bar')).toBeTruthy();
  });

  it('hides the splash screen once fonts are loaded', async () => {
    const hideAsyncMock = require('expo-splash-screen').hideAsync;

    render(<RootLayout />);

    // Ensure splash screen hides after font loading
    await waitFor(() => {
      expect(hideAsyncMock).toHaveBeenCalled();
    });
  });
});

**Comment by DAMSCROW** [comment]
// Metro configuration file
const { getDefaultConfig } = require("expo/metro-config");
const { resolve } = require("node:path");
const { withNativewind } = require("nativewind/metro");
const { makeMetroConfig } = require("@rnx-kit/metro-config");
const exclusionList = require("metro-config/src/defaults/exclusionList");

// Find the project and workspace directories
const projectRoot = __dirname;
const monorepoRoot = resolve(projectRoot, "../..");

const expoConfig = getDefaultConfig(projectRoot);

const config = makeMetroConfig({
  ...expoConfig,
  resolver: {
    ...expoConfig.resolver,
    nodeModulesPaths: [resolve(projectRoot, "node_modules"), resolve(monorepoRoot, "node_modules")],
    assetExts: expoConfig.resolver.assetExts.filter((ext) => ext !== "svg"),
    sourceExts: [...expoConfig.resolver.sourceExts, "svg"],
    blacklistRE: exclusionList([/.*\.test\.tsx?$/]), // Exclude test files from the bundle
    unstable_enableSymlinks: true,
  },
  transformer: {
    ...expoConfig.transformer,
    babelTransformerPath: require.resolve("react-native-svg-transformer/expo"),
  },
  watchFolders: [monorepoRoot],
});

module.exports = withNativewind(config, { input: ["global.css"] });

**Comment by DAMSCROW** [comment]
The Exceptional Journey of Team EMBER: From POC to MVP

From its inception as a Proof of Concept (POC), Team EMBER has consistently demonstrated unparalleled excellence and collaboration. With the successful acceptance of the POC, the project has now transitioned into the Minimum Viable Product (MVP) stage—a testament to the team’s hard work, adaptability, and dedication to delivering quality outcomes.

Team EMBER has embraced top-edge technologies in the AWS and broader cloud ecosystem, continuously upgrading their skillsets to stay ahead in a rapidly evolving tech landscape. This proactive approach has not only ensured the team’s technical competence but has also fostered an environment of mutual learning and growth.

A culture of collaboration and rigorous peer reviews has been at the heart of EMBER’s success. The team consistently supports each other, ensuring that every output meets the highest quality standards while maintaining streamlined workflows without delays or disruptions. Their unwavering focus on excellence has led to smooth operations and effective problem-solving, setting a benchmark for others.

The project itself has become a hallmark of success, serving as a template for incoming client projects. EMBER’s ability to deliver innovative, reliable, and scalable solutions has earned the team recognition as a trailblazer, driving confidence and inspiration across the organization.

Team EMBER is not just delivering a project—they are defining the standards for success. Their story stands as a shining example of what can be achieved when passion, skill, and teamwork align.

**Comment by DAMSCROW** [comment]
The Exceptional Journey of Team EMBER: From POC to MVP

From its inception as a Proof of Concept (POC), Team EMBER has consistently demonstrated unparalleled excellence and collaboration. With the successful acceptance of the POC, the project has now transitioned into the Minimum Viable Product (MVP) stage—a testament to the team’s hard work, adaptability, and dedication to delivering quality outcomes.

Team EMBER has embraced top-edge technologies in the AWS and broader cloud ecosystem, continuously upgrading their skillsets to stay ahead in a rapidly evolving tech landscape. This proactive approach has not only ensured the team’s technical competence but has also fostered an environment of mutual learning and growth.

A culture of collaboration and rigorous peer reviews has been at the heart of EMBER’s success. The team consistently supports each other, ensuring that every output meets the highest quality standards while maintaining streamlined workflows without delays or disruptions. Their unwavering focus on excellence has led to smooth operations and effective problem-solving, setting a benchmark for others.

The project itself has become a hallmark of success, serving as a template for incoming client projects. EMBER’s ability to deliver innovative, reliable, and scalable solutions has earned the team recognition as a trailblazer, driving confidence and inspiration across the organization.

Team EMBER is not just delivering a project—they are defining the standards for success. Their story stands as a shining example of what can be achieved when passion, skill, and teamwork align.

**Comment by DAMSCROW** [comment]
console.log("Blacklist Regex:", /.*\/_tests_\/.*\.test\.(js|ts|jsx|tsx)$/); // Log regex for validation
